lib/internal/worker.js

```diff
@@ -527,6 +543,79 @@ function eventLoopUtilization(util1, util2) {
   );
 }
 
+function setConnectionsListener(fn) {
+  if (isMainThread && !mainPortWasSetup) {
+    setupMainPort();
+    mainPortWasSetup = true;
+  }
+
+  if (typeof fn === 'undefined' || fn === null) {
+    connectionsListener = null;
+    return;
+  }
+
+  validateFunction(fn, 'fn');
+  connectionsListener = fn;
+}
+
+function processConnectionRequest(request) {
+  const status = new Int32Array(request.memory);
+
+  if (connectionsListener?.(request.sender, request.port, request.data) !== true) {
+    AtomicsStore(status, WORKER_MESSAGING_RESULT_INDEX, 1);
+    AtomicsStore(status, WORKER_MESSAGING_STATUS_INDEX, 1);
+    AtomicsNotify(status, WORKER_MESSAGING_STATUS_INDEX);
+    return;
+  }
+
+  AtomicsStore(status, WORKER_MESSAGING_RESULT_INDEX, 0);
+  AtomicsStore(status, WORKER_MESSAGING_STATUS_INDEX, 1);
+  AtomicsNotify(status, WORKER_MESSAGING_STATUS_INDEX);
+}
+
+function connect(target, data) {
+  if (target === threadId) {
+    throw new ERR_WORKER_SAME_THREAD();
+  }
+
+  // Create a shared array to exchange the status and the result
+  const memory = new SharedArrayBuffer(WORKER_MESSAGING_SHARED_DATA);
+  const status = new Int32Array(memory);
+
+  // Create the channel and send it to the other thred
+  const { port1, port2 } = new MessageChannel();
+  sendToWorker(target, { type: messageTypes.CONNECT, sender: threadId, port: port2, memory, data }, [port2]);
```

Comment:
Is there also a disconnect implicitly happening when one of the threads that have a channel ends?

Reply:
Can you please clarify?

Reply:
I was thinking if the existence of this connection (`MessageChannel`) might cause the thread to not exit as before this change because there is a `port.on()` subscription being registered somewhere? Please ignore if that's not the case

Reply:
The `sendToWorker` uses the same port that Node uses to have parent and children talk, so it should not add an additional ref.